### PR TITLE
ETCM-168: Fix stalled enrollments

### PR DESCRIPTION
**Pull request by aakoshh**
Sometimes `enroll` seems to never finish. After debugging I think it's the `bondNeighbors`.

I couldn't see anything obvious in the code, but it's conceivable that if there are multiple bonding attempts then a deferred somehow got stuck in the map. Instead of waiting on it forever it now returns a result after a timeout.

Also limited the number of parallel bonding attempts to `alpha`.

### CODE REVIEW

File: scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala [hunk 1]
```diff
@@ -775,18 +786,19 @@ object DiscoveryService {
...
-            .flatMap(bondNeighbors)
-            .flatMap { newNeighbors =>
-              val newClosest = (closest ++ newNeighbors).take(config.kademliaBucketSize)
-              val newAsked = asked ++ contacts
-              loop(local, newClosest, newAsked)
-            }
+        if (contacts.isEmpty) {
+          Task(logger.debug(s"Lookup finished for $target after asking ${asked.size} nodes.")).as(closest)
+        } else {
+          Task(logger.debug(s"Lookup for $target contacting ${contacts.size} nodes.")) >>
```
Comment (KonradStaniec) [inline scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala]: Maybe we could also log how many nodes we already asked ?

File: scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala [hunk 2]
```diff
@@ -775,18 +786,19 @@ object DiscoveryService {
...
+            Task
+              .parTraverseUnordered(contacts)(fetchNeighbors)
+              .map(_.flatten.distinct)
+              .flatMap(bondNeighbors)
+              .flatMap { newNeighbors =>
+                val newClosest = (closest ++ newNeighbors).take(config.kademliaBucketSize)
```
Comment (KonradStaniec) [inline scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala]: Maybe we could log how many new neighbours we had found ?

**Review by KonradStaniec — APPROVED** [review]

File: scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala (hunk 1 above)
Comment (aakoshh) [reply scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala]: Sure thing.

File: scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala (hunk 2 above)
Comment (aakoshh) [reply scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala]: :+1:

File: scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala (hunk 1 above)
Comment (aakoshh) [reply scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala]: I thought we could log the addresses too but it's a bit verbose.

**Pull request merged**